ambari-server/src/main/java/org/apache/ambari/server/orm/entities/ClusterServiceEntity.java

```diff
@@ -29,46 +29,46 @@
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
 import javax.persistence.JoinColumn;
-import javax.persistence.JoinColumns;
 import javax.persistence.ManyToOne;
 import javax.persistence.NamedQueries;
 import javax.persistence.NamedQuery;
 import javax.persistence.OneToMany;
 import javax.persistence.OneToOne;
+import javax.persistence.Table;
 import javax.persistence.TableGenerator;
+import javax.persistence.UniqueConstraint;
 
-@javax.persistence.IdClass(ClusterServiceEntityPK.class)
-@javax.persistence.Table(name = "clusterservices")
 @NamedQueries({
   @NamedQuery(name = "clusterServiceById", query =
     "SELECT clusterService " +
       "FROM ClusterServiceEntity clusterService " +
-      "JOIN clusterService.serviceGroupEntity serviceGroup " +
-      "WHERE clusterService.serviceId=:serviceId " +
-      "AND  serviceGroup.serviceGroupId=:serviceGroupId " +
-      "AND serviceGroup.clusterId=:clusterId"),
-   @NamedQuery(name = "clusterServiceByName", query =
-     "SELECT clusterService " +
-      "FROM ClusterServiceEntity clusterService " +
-       "JOIN clusterService.serviceGroupEntity serviceGroup " +
-        "JOIN clusterService.clusterEntity clusterEntity " +
-       "WHERE clusterService.serviceName=:serviceName " +
-       "AND  serviceGroup.serviceGroupName=:serviceGroupName " +
-        "AND clusterEntity.clusterName=:clusterName")
+      "WHERE clusterService.serviceId=:serviceId "),
+  @NamedQuery(name = "clusterServiceByName", query =
+    "SELECT clusterService " +
+    "FROM ClusterServiceEntity clusterService " +
+    "JOIN clusterService.serviceGroupEntity serviceGroup " +
+    "JOIN clusterService.clusterEntity clusterEntity " +
+    "WHERE clusterService.serviceName=:serviceName " +
+    "AND  serviceGroup.serviceGroupName=:serviceGroupName " +
+    "AND clusterEntity.clusterName=:clusterName")
 })
```

Comment:
Instead of doing the `JOINS` yourself, can you do

```
SELECT clusterService FROM ClusterServiceEntity clusterService WHERE 
  clusterService.serviceName=:serviceName AND 
  clusterService.serviceGroupEntity.serviceGroupName=:serviceGroupName AND 
  clusterService.clusterEntity.clusterName:=clusterName
```

Reply:
Yep. Updated this part.